src/app/resources/iis_scripts/admin_api_util.ps1

```diff
@@ -0,0 +1,126 @@
+param(
+    [Parameter(Mandatory=$true)]
+    [ValidateSet('install', 'ensure-permission')]
+    [string]
+    $command,
+
+    [string]
+    $downloadFrom,
+
+    ## Not doing anything with this parameter yet
+    [string]
+    $sessionId,
+
+    [string]
+    $serviceName = "Microsoft IIS Administration",
+
+    [string]
+    $iisAdminOwners = "IISAdminAPIOwners"
+)
+
+$ErrorActionPreference = "Stop"
+
+######################## Param checks ###############################
+$install = $command -eq 'install'
+
+if ($install -and $(!$downloadFrom)) {
+    throw "Missing download location for install command"
+}
+
+######################## Utilities ###############################
+
+function WaitForServerToStart($service) {
+    $waitPeriod = 1
+    $remainingTime = 600
+
+    while (!(Get-Service $service -ErrorAction SilentlyContinue | Where-Object {$_.Status -eq "Running"})) {
+        Start-Sleep -Seconds $waitPeriod
+        $remainingTime -= $waitPeriod
+        if ($remainingTime -lt 0) {
+            throw "timeout waiting for service to start"
+        }
+    }
+}
+
+function InstallAPI {
+    $installer = Join-Path $env:TEMP iis-administration-setup.exe
+    Invoke-WebRequest $downloadFrom -OutFile $installer
+    & $installer /s
+    WaitForServerToStart $serviceName
+}
+
+function EnsureGroup($group) {
+    if (!(Get-LocalGroup -Name $group -ErrorAction SilentlyContinue)) {
+        New-LocalGroup -Name $group | Out-Null
+    }
+}
+
+function EnsureMember($group, $userOrGroup) {
+    if (!(Get-LocalGroupMember -Group $group -Member $userOrGroup -ErrorAction SilentlyContinue)) {
+        Add-LocalGroupMember -Group $group -Member $userOrGroup | Out-Null
+    }
+}
+
+####################### Main script ################################
+
+## TODO: Consider creating a lock around this script because it is not concurrency safe
+
+if ($install) {
+    InstallAPI
+}
+
+$service = Get-WmiObject win32_service | Where-Object {$_.name -eq $serviceName}
+if ($service.StartInfo.EnvironmentVariables -and $service.StartInfo.EnvironmentVariables["USE_CURRENT_DIRECTORY_AS_ROOT"] -and $service.StartInfo.WorkingDirectory) {
+    $workingDirectory = $service.StartInfo.WorkingDirectory
+} else {
+    $proc = Get-Process -id $service.ProcessId
+    $workingDirectory = Split-Path ($proc.Modules | Where-Object { $_.ModuleName -eq "Microsoft.IIS.Administration.dll" }).FileName
+}
+
+$configLocation = [System.IO.Path]::Combine($workingDirectory, "config", "appsettings.json")
+$config = Get-Content -Raw -Path $configLocation | ConvertFrom-Json
+
+$user = $(whoami)
+EnsureGroup $iisAdminOwners
+EnsureMember $iisAdminOwners $user
+
+if (!$config.security.users.owners.Contains($iisAdminOwners)) {
+    foreach ($member in $config.security.users.owners) {
+        EnsureMember $iisAdminOwners $member
+    }
+    $config.security.users.owners = @( $iisAdminOwners )

```

Comment:
Owner's could already have members right? Shouldn't $iisAdminOwners be appended to the existing list of owners rather than replace it.

Reply:
My current thinking is that if WAC get involved after Admin API is installed, we would just manage permission with local group. Note that there's a loop adding all the members in the iisAdminOwners group

Reply:
I am trying to keep a unified place to manage access. Consider this scenario: User A installed the program prior to using WAC. Now A is a owner in the config file but he has no write access to the config file as intended.
Now A uses WAC and the script is run, we still have to add user A to the iisAdminOwners for him to add iisAdminOwners into the owner group of the config file. Now this user's access would have been defined in 2 places, the config file says A has access. Also config file says iisAdminOwner has access, which he is.
Seems to be a problem in maintenance.